Fixed missing char + added some more function documentation

TheOtterlord opened the pull request:
- Fixes #5
- Adds some more function docs
- Adds a function call to example file

TheOtterlord committed:
Fix missing first char + test uses main + docs updates

- Fixed missing character from issue #5
- Renamed test_function_call to test_function_call_without_args and renamed test_1 to main
- Added code to example.gpl
- Added some docs for functions

TheOtterlord committed:
Merge remote-tracking branch 'upstream/master'

mjarkk commented on src/lib/parser.rs:
```diff
@@ -150,7 +150,7 @@ impl Parser {
     }
 
-    let mut char_count: usize = 0;
+    let mut char_count: usize = 1;
```
This won't work because a few lines after here we do `match option.as_bytes().get(char_count) {` and that will return invalid relative characters.  
below the `while` we can better do:
```
self.index -= char_count + 1;
```

theotterlord replied:
Ah, this is much better. I'll make the changes now

mjarkk commented on src/example.gpl:
```diff
@@ -1,3 +1,5 @@
 
 fn take_args(a int, b int) {}
+
+fn main() { empty() }
```
Maybe we should add this file to the .gitignore so we can freely test code

theotterlord replied:
Yeah, this would make things much easier. 
Since we now have some documentation and a roadmap, this file is not needed for the original purposes.

theotterlord replied:
I've added a filter for all gpl files in src. This will help when we need to test multi-file features

theotterlord replied:
I've added a filter for all gpl files in src. This will help when we need to test multi-file features

mjarkk reviewed (changes requested):
Found a few small issues

TheOtterlord committed:
Modify char_count

TheOtterlord committed:
Add gpl file extension to gitignore

Allows for more than one file when testing. Will be useful for library testing etc...

mjarkk commented:
Great! i'll merge the pr

mjarkk commented:
maybe something next could be adding support for comments.  
I thought they would just be something like `//` for 1 line comments and `/* */` for multi line comments

theotterlord commented:
Yeah, most languages use this syntax so it will help people migrate over. It also allows us to start documenting our tests & examples.

mjarkk commented:
I think this could be best implemented in the `next_char` function because this needs to be everywhere possible.  
Although if we ever want to support code .gpl file formatting we might want to remember some comment context but we can always add that later.

theotterlord commented:
For now, best to just implement the ability to use comments. And yes, I think that `next_char` is a good place to put it.
